docker-entrypoint.sh

```diff
@@ -35,12 +35,26 @@ drop_privs_cmd() {
     fi
 }
 
+copy_plugins_if_required() {
+  echo "Enabling required built-in plugins"
+  for TARGET_PLUGIN in $(echo "$ENABLE_BUILT_IN_PLUGINS" | grep -o -e "[^;]*"); do
+    echo "Moving $TARGET_PLUGIN to plugin directory"
+    if cp "${FLINK_HOME}/opt/${TARGET_PLUGIN}" "${FLINK_HOME}/plugins"; then
+      echo "Successfully enabled $TARGET_PLUGIN"
+    else
+      echo "Failed to enable $TARGET_PLUGIN. Exiting."
+      exit 1
+    fi
+  done
+}
+
 if [ "$1" = "help" ]; then
     echo "Usage: $(basename "$0") (jobmanager|taskmanager|help)"
     exit 0
 elif [ "$1" = "jobmanager" ]; then
     shift 1
     echo "Starting Job Manager"
+    copy_plugins_if_required
```

Comment:
I'm not really familiar with Docker, but please double-check that this is not modifying the base image. We need proper isolation between two different invocations of docker with different `ENABLE_BUILT_IN_PLUGINS`, especially for s3 where two plugins serve the same prefix. 
My guess is that the entrypoint is not modifying the layer and all is good.

Reply:
Yep, the entrypoint **script** will be changed in the base images pushed to Docker Hub, but the entrypoint is only invoked at runtime. Someone using the container without the environment variable set will not see any changes whatsoever.